scvi/model/base/_training_mixin.py

```diff
@@ -98,3 +98,100 @@ def train(
+            `np.min([round((20000 / n_cells) * 400), 400])`
+        %(param_accelerator)s
+        %(param_devices)s
+        train_size
+            Size of training set in the range [0.0, 1.0].
+        validation_size
+            Size of the test set. If `None`, defaults to 1 - `train_size`. If
+            `train_size + validation_size < 1`, the remaining cells belong to a test set.
+        shuffle_set_split
+            Whether to shuffle indices before splitting. If `False`, the val, train, and test set are split in the
+            sequential order of the data according to `validation_size` and `train_size` percentages.
+        load_sparse_tensor
+            `EXPERIMENTAL` If ``True``, loads data with sparse CSR or CSC layout as a
+            :class:`~torch.Tensor` with the same layout. Can lead to speedups in data transfers to
+            GPUs, depending on the sparsity of the data.
+        batch_size
+            Minibatch size to use during training.
+        early_stopping
+            Perform early stopping. Additional arguments can be passed in `**kwargs`.
+            See :class:`~scvi.train.Trainer` for further options.
+        datasplitter_kwargs
+            Additional keyword arguments passed into :class:`~scvi.dataloaders.ContrastiveDataSplitter`.
+        plan_kwargs
+            Keyword args for :class:`~scvi.train.TrainingPlan`. Keyword arguments passed to
+            `train()` will overwrite values present in `plan_kwargs`, when appropriate.
+        **trainer_kwargs
+            Other keyword args for :class:`~scvi.train.Trainer`.
+        """
+        if max_epochs is None:
+            max_epochs = get_max_epochs_heuristic(self.adata.n_obs)
+
+        plan_kwargs = plan_kwargs or {}
+        datasplitter_kwargs = datasplitter_kwargs or {}
+
+        data_splitter = self._data_splitter_cls(
+            self.adata_manager,
+            background_indices=background_indices,
+            target_indices=target_indices,
+            train_size=train_size,
+            validation_size=validation_size,
+            batch_size=batch_size,
+            shuffle_set_split=shuffle_set_split,
+            distributed_sampler=use_distributed_sampler(
+                trainer_kwargs.get("strategy", None)
+            ),
+            load_sparse_tensor=load_sparse_tensor,
+            **datasplitter_kwargs,
+        )
+        training_plan = self._training_plan_cls(self.module, **plan_kwargs)
+
+        es = "early_stopping"
+        trainer_kwargs[es] = (
+            early_stopping if es not in trainer_kwargs.keys() else trainer_kwargs[es]
+        )
+        runner = self._train_runner_cls(
+            self,
+            training_plan=training_plan,
+            data_splitter=data_splitter,
+            max_epochs=max_epochs,
+            accelerator=accelerator,
+            devices=devices,
+            **trainer_kwargs,
+        )
+        return runner()
```

Comment:
Preferable if this code lives directly in `ContrastiveVI` as the train method is not used by any other model (at least for now).

Reply:
Sounds good! I moved the code into `ContrastiveVI`.